src/qforte/quantum_computer.h

```diff
@@ -76,24 +76,49 @@ class QuantumCircuit {
     /// return a vector of gates
     const std::vector<QuantumGate>& gates() const { return gates_; }
 
+    /// return a vector of reversed gates
+    const std::vector<QuantumGate>& reversed_gates() const { return rev_copy_; }
+
+    /// reset the circuit with a new set of parameters
+    void set_parameters(const std::vector<double>& params);
+
+    /// set the reversed gates 
```

Comment:
Why needed? This can be buried in `reversed_gates()` so that is automatically called.